### PR TITLE
Freeze objects by default, stdlib improvements.

### PR DESCRIPTION
**Pull request by olonho**

### CODE REVIEW

File: runtime/src/main/kotlin/konan/Annotations.kt [hunk 1]
```diff
@@ -101,18 +71,26 @@ public annotation class FixmeReflection
...
 public annotation class PointsTo(vararg val onWhom: Int)
 
 /**
- * Need to be fixed because of header/impl notation
+ * Top level variable or object is thread local, and so could be mutable.
  */
-public annotation class FixmeMultiplatform
+@Target(AnnotationTarget.FIELD, AnnotationTarget.CLASS)
+@Retention(AnnotationRetention.BINARY)
+annotation class ThreadLocal
```
Comment (SvyatoslavScherbina) [inline runtime/src/main/kotlin/konan/Annotations.kt]: Is it still required?

File: runtime/src/main/kotlin/konan/worker/Freezing.kt [hunk 2]
```diff
@@ -26,7 +26,8 @@ public class FreezingException() : RuntimeException()
 /**
  * Exception thrown whenever we attempt to mutate frozen objects.
  */
-public class InvalidMutabilityException() : RuntimeException()
+public class InvalidMutabilityException(where: Any) :
+        RuntimeException("mutation attempt of frozen $where (hash is 0x${(where as Any).hashCode().toString(16)})")
```
Comment (SvyatoslavScherbina) [inline runtime/src/main/kotlin/konan/worker/Freezing.kt]: What's the purpose of casting `Any` to `Any`?

File: runtime/src/main/kotlin/konan/Annotations.kt (hunk 1 above)
Comment (olonho) [reply runtime/src/main/kotlin/konan/Annotations.kt]: Well, without it we cannot locally disable freezing on objects. I can write big fat warning about instability of such annotation.

File: runtime/src/main/kotlin/konan/worker/Freezing.kt (hunk 2 above)
Comment (olonho) [reply runtime/src/main/kotlin/konan/worker/Freezing.kt]: Maybe none, I just didn't want virtual dispatch.

File: runtime/src/main/kotlin/konan/Annotations.kt (hunk 1 above)
Comment (SvyatoslavScherbina) [reply runtime/src/main/kotlin/konan/Annotations.kt]: What's the purpose of introducing feature that is already deprecated and is not used?

**Pull request merged**